### PR TITLE
Registering a default response when no match

### LINKED ISSUE #25: Registering a default response when no match
I've been trying to figure out how I can make my bot respond with a helpful message when no message is matched.

Naively I've tried the following approach which results in both handlers to be executed
```
b.Respond("foo", b.FooHandler)
b.Respond(".*", b.HelpHandler)
```

Unless I'm missing something, there does not seem to be a way to register a default handler that would only be executed if no other handler matches the message.

There also does not appear to be a way to stop processing handlers after the `FooHandler` is executed.

### PR DESCRIPTION
**Pull request by fgrosse**
Fixes #25 

- improve documentation about how multiple matching event handlers are executed
- add a way for handlers to prevent execution of later event handlers
- use this in `Bot.Respond(…)` so only the first matching handler is executed
- unit test it all

### CODE REVIEW

File: bot.go [hunk 1]
```diff
@@ -227,7 +227,12 @@ func (b *Bot) Run() error {
 //
 // If you need complete control over the regular expression, e.g. because you
 // want the patter to match only a substring of the message but not all of it,
-// you can use Bot.RespondRegex(…).
+// you can use Bot.RespondRegex(…). For even more control you can also directly
+// use Brain.RegisterHandler(…) with a function that accepts ReceiveMessageEvent
+// instances.
+//
+// If multiple matching patterns are registered, only the first registered
```
Comment (jfchevrette) [inline bot.go]: I think this should say "... only the first matched handler is executed" (not the first registered)

**Comment by jfchevrette** [comment]
Also if I understand the change corretly, this changes the current behavior which is to continue processing handlers set via `Bot.Respond()` or `Bot.RespondRegex()`

If that's the case, then would that warrant a .y version bump and a breaking change note in the Changelog?

File: bot.go (hunk 1 above)
Comment (fgrosse) [reply bot.go]: Thanks for the feedback, gonna update this bit of the documentation to clarify the intended behavior :+1:

**Comment by fgrosse** [comment]
Regarding breaking changes, that's why Joe is still in alpha (i.e. there is no version 1 yet)

> **THIS SOFTWARE IS STILL IN ALPHA AND THERE ARE NO GUARANTEES REGARDING API STABILITY YET.**

The changes will likely be released this week via v0.10.0, currently its still in the _Unreleased_ section of the `CHANGELOG.md`

File: bot.go (hunk 1 above)
Comment (fgrosse) [reply bot.go]: How about:

```
// If multiple matching patterns are registered, only the first matching handler
// is executed (ordered by sequence of handler registration).
```

I still want to make clear that it is important in what order you register your handlers but I'n not sure the changed comment is actually simpler to read while it still contains the same info. WDYT?

**Pull request merged**